docs/ImportantPythonFunctions.md

```diff
@@ -0,0 +1,180 @@
+graph_def = helper.make_graph(
+    [node_def],
+    "test-model",
+    [X],
+    [Y],
+)
+
+# Create the model (ModelProto)
+model_def = helper.make_model(graph_def, producer_name='onnx-example')
+
+print('The model is:\n{}'.format(model_def))
+onnx.checker.check_model(model_def)
+print('The model is checked!')
+```
+Runnable IPython notebooks:
+- [make_model.ipynb](https://github.com/onnx/onnx/tree/master/onnx/examples/make_model.ipynb)
+- [Protobufs.ipynb](https://github.com/onnx/onnx/tree/master/onnx/examples/Protobufs.ipynb)
+
+## Checking an ONNX Model
+```python
+import onnx
+import os
+
+# Preprocessing: load the ONNX model
+model_path = os.path.join(os.path.dirname('model.onnx')
+onnx_model = onnx.load(model_path)
+
+print('The model is:\n{}'.format(onnx_model))
+
+# Check the model
+onnx.checker.check_model(onnx_model)
+print('The model is checked!')
+```
+Runnable IPython notebooks:
+- [check_model.ipynb](https://github.com/onnx/onnx/tree/master/onnx/examples/check_model.ipynb)
+
+## Optimizing an ONNX Model
+```python
+import onnx
+import os
+from onnx import optimizer
+
+# Preprocessing: load the model contains two transposes.
+model_path = os.path.join('path/to/the/model')
+original_model = onnx.load(model_path)
+original_model_str = original_model.SerializeToString()
+
+print('The model before optimization:\n{}'.format(original_model))
+
+# A full list of supported optimization passes can be found here:
+# https://github.com/onnx/onnx/blob/master/onnx/optimizer.py#L21
+pass_list = ['fuse_consecutive_transposes']
+
+# Apply the optimization on the original serialized model
+optimized_model = onnx.load_from_string(optimizer.optimize(original_model_str, pass_list))
+
+print('The model after optimization:\n{}'.format(optimized_model))
+```
+Runnable IPython notebooks:
+- [optimize_onnx.ipynb](https://github.com/onnx/onnx/tree/master/onnx/examples/optimize_onnx.ipynb)
+
+## Running Shape Inference on an ONNX Model
+```python
+import onnx
+from onnx import helper, shape_inference
+from onnx import TensorProto
+
+
+# Preprocessing: create a model with two nodes, Y's shape is unknown
+node1 = helper.make_node('Transpose', ['X'], ['Y'], perm=[1, 0, 2])
+node2 = helper.make_node('Transpose', ['Y'], ['Z'], perm=[1, 0, 2])
+
+graph = helper.make_graph(
+    [node1, node2],
+    'two-transposes',
+    [helper.make_tensor_value_info('X', TensorProto.FLOAT, (2, 3, 4))],
+    [helper.make_tensor_value_info('Z', TensorProto.FLOAT, (2, 3, 4))],
+)
+
+original_model = helper.make_model(graph, producer_name='onnx-examples')
+
+# Check the model and print Y's shape information
+onnx.checker.check_model(original_model)
+print('Before shape inference, the shape info of Y is:\n{}'.format(original_model.graph.value_info))
+
+# Apply shape inference on the model
+original_model_str = original_model.SerializeToString()
+inferred_model_str = shape_inference.infer_shapes(original_model_str)
+inferred_model = onnx.load_from_string(inferred_model_str)
```

Comment:
same here - hide serialization inside the library

Reply:
Yep, I will do that in a different PR. And then rebase the introduction doc.